google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/EnhancedBigtableStub.java

```diff
@@ -120,65 +123,89 @@
 
   public static EnhancedBigtableStub create(EnhancedBigtableStubSettings settings)
       throws IOException {
-    ClientContext clientContext = ClientContext.create(settings);
+    settings = finalizeSettings(settings, Tags.getTagger(), Stats.getStatsRecorder());
 
-    return new EnhancedBigtableStub(
-        settings, clientContext, Tags.getTagger(), Stats.getStatsRecorder());
+    return new EnhancedBigtableStub(settings, ClientContext.create(settings));
   }
 
-  @InternalApi("Visible for testing")
-  public EnhancedBigtableStub(
-      EnhancedBigtableStubSettings settings,
-      ClientContext clientContext,
-      Tagger tagger,
-      StatsRecorder statsRecorder) {
-    this.settings = settings;
+  public static EnhancedBigtableStubSettings finalizeSettings(
+      EnhancedBigtableStubSettings settings, Tagger tagger, StatsRecorder stats)
+      throws IOException {
+    EnhancedBigtableStubSettings.Builder builder = settings.toBuilder();
+
+    if (settings.isRefreshingChannel()) {
```

Comment:
Can we add some tests around this to ensure the proper values are set in the settings?

Reply:
I added a test for testChannelPrimerConfigured, which should cover the behavior of it